loadtest/control/simulcontroller/actions.go

```diff
@@ -1831,3 +1832,156 @@ func (c *SimulController) createPostReminder(u user.User) control.UserActionResp
+	}
+	var postID string
+	for _, p := range posts {
+		// Found a permalink.
+		if index := strings.Index(p.Message, "/pl/"); index != -1 {
+			// If there are multiple permalinks found, we will click on the last one
+			// in the channel. This naturally leads to an effect of having clicked all
+			// permalinks which emulates organic behavior.
+			start := index + len("/pl/")
+			idLen := 26 // All IDs are always 26-char long
+			postID = p.Message[start : start+idLen]
+		}
+	}
+	if postID == "" {
+		return control.UserActionResponse{Info: "no permalink found"}
+	}
+
+	// We fetch the post thread.
+	postIds, _, err := u.GetPostThreadWithOpts(postID, "", model.GetPostsOptions{
+		CollapsedThreads: true,
+		Direction:        "down",
+		PerPage:          25,
+	})
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	oldestPost, err := u.Store().Post(postIds[0])
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+
+	ch, err := u.Store().Channel(oldestPost.ChannelId)
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	// We fetch the channel if it's not present in state.
+	// This is similar to what the webapp does.
+	if ch == nil {
+		if err := u.GetChannel(oldestPost.ChannelId); err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+		// We fetch the channel data from store again.
+		ch, err = u.Store().Channel(oldestPost.ChannelId)
+		if err != nil {
+			return control.UserActionResponse{Err: control.NewUserError(err)}
+		}
+	}
+
+	// We check if membership data is already there. If not,
+	// we call getChannelMember, and add the user to the channel
+	// if it's not a private channel.
+	cm, err := u.Store().ChannelMember(ch.Id, u.Store().Id())
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
+	if cm.UserId == "" {
+		err := u.GetChannelMember(ch.Id, u.Store().Id())
+		if err != nil {
+			// Resorting to string matching, because the API doesn't return the
+			// actual Response object.
+			if !strings.Contains(err.Error(), "You do not have the appropriate permissions") {
+				return control.UserActionResponse{Err: control.NewUserError(err)}
+			}
+
+			if ch.Type != model.ChannelTypePrivate {
+				if err := u.AddChannelMember(ch.Id, u.Store().Id()); err != nil {
+					return control.UserActionResponse{Err: control.NewUserError(err)}
+				}
+			}
+		}
+	}
+
+	currentTeam, err := u.Store().CurrentTeam()
+	if err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	} else if currentTeam == nil {
+		return control.UserActionResponse{Err: control.NewUserError(errors.New("current team should be set"))}
+	}
+
+	if _, err := u.GetChannelsForTeamForUser(currentTeam.Id, u.Store().Id(), true); err != nil {
+		return control.UserActionResponse{Err: control.NewUserError(err)}
+	}
```

Comment:
Does the webapp ask for deleted channels as well?

Reply:
Hmm .. let me confirm that. I didn't check the `is_deleted` flag.

Reply:
It doesn't.